docs/src/references/c8y-firmware-management.md

```diff
@@ -0,0 +1,214 @@
+which are configured using the tedge config settings `mqtt.client.host` or `mqtt.client.port` for MQTT
+and `http.address` and `http.port` for HTTP.
+
+
+### Child devices declaring firmware management support
+
+But for child devices, the operation files must be created under `/etc/tedge/operations/c8y/$CHILD_DEVICE_ID`.
+These files are not created by the plugin itself, but must be created by the child device connector or
+by any other means for each child device as follows:
+
+
+```shell
+$ tree /etc/tedge/operations/c8y
+/etc/tedge/operations/c8y
+|-- child-1
+|   `-- c8y_Firmware
+`-- child-2
+    `-- c8y_Firmware
+```
+
+Using the file watcher mechanism, the c8-mapper picks up the creation of these operation files
+and report them as supported operations for the child device as follows:
+
+```shell
+$ tedge mqtt sub 'c8y/s/us/#'
+[c8y/s/us/child-1] 114,c8y_Firmware
+[c8y/s/us/child-2] 114,c8y_Firmware
+```
+
+
+### The child device connector handling firmware update requests from thin-edge device
+
+When the plugin receives a firmware update request file for a child device,
+it downloads the firmware file, caches it to be reused on other child devices as well
+and exposes it to the child device via its file-transfer service. 
+It also notifies the cloud on the progress of this firmware update operation
+as and when it gets status updates from the child device.
+
+The following diagram captures the required interactions between all relevant parties:
+
+```mermaid
+sequenceDiagram
+    participant C8Y Cloud
+    participant C8Y Firmware Plugin
+    participant Tedge Agent
+    participant Child Device Connector
+    
+
+        C8Y Cloud->>C8Y Firmware Plugin: MQTT: c8y_Firmware request with `child-id` and `c8y.url` for the updated firmware file
+        C8Y Firmware Plugin->>C8Y Cloud: Download the firmware file from `c8y.url` to firmware cache
+        C8Y Firmware Plugin->>Tedge Agent: Symlink the cached firmware file to file-transfer repository and generate a `tedge.url` for it
+        C8Y Firmware Plugin->>Child Device Connector: MQTT: firmware_update request to `child-id` with `tedge.url`
+
+        Child Device Connector ->> C8Y Firmware Plugin: MQTT: firmware_update response with operation status: "executing" 
+        C8Y Firmware Plugin ->> C8Y Cloud: MQTT: Update c8y_Firmware operation status to "EXECUTING"
+        Child Device Connector->>Tedge Agent: HTTP: Download the firmware file from `tedge.url`
+        Child Device Connector->>Child Device Connector: Apply downloaded firmware file
+        Child Device Connector ->> C8Y Firmware Plugin: MQTT: firmware_update response with operation status: "successful" 
+
+        C8Y Firmware Plugin ->> C8Y Cloud: MQTT: Update c8y_Firmware operation status to "SUCCESSFUL"
+        C8Y Firmware Plugin ->> C8Y Firmware Plugin: Remove the symlink from file-transfer repository but keep the cached firmware copy for reuse
```

Comment:
I don't think it removes symlink... instead deleting a firmware info file under `FIRMWARE_OP_PATH`.

Reply:
We should be removing those symlinks as it makes it easier for the device admin to cleanup the firmware cache, when needed by just looking at the files that don't have any symlinks pointed to it.